lib/docker/executeDockerBuild.ts

```diff
@@ -40,45 +41,69 @@ import * as os from "os";
 import * as path from "path";
 import { cleanImageName } from "./name";
 
-/**
- * Options to configure the Docker image build
- */
-export interface DockerOptions {
-
+export interface DockerRegistry {
     /**
-     * Provide the image tag for the docker image to build
+     * Push Url for this registry
      */
-    dockerImageNameCreator?: DockerImageNameCreator;
+    url: string;
 
     /**
-     * True if the docker image should be pushed to the registry
+     * Should this registry be displayed with the goal status after a push? Default true.
      */
-    push?: boolean;
+    display?: boolean;
 
     /**
-     * Optional registry to push the docker image too.
-     * Needs to set when push === true
+     * Display Url - ie the url humans can go to
+     * assumes <url>/<image>
      */
-    registry?: string;
+    displayUrl?: string;
 
     /**
-     * Optional user to use when pushing the docker image.
-     * Needs to set when push === true
+     * If specified, this will replace the label version details (eg <image><:version>)
+     * For example, for Dockerhub the correct value would be `/tags`, with a displayUrl set
+     * to https://hub.docker.com/r/<user/org>; will result in:
+     * https://hub.docker.com/r/<user/org>/<image>/tags as the link URL
+     *
      */
-    user?: string;
+    displayBrowsePath?: string;
 
     /**
-     * Optional password to use when pushing the docker image.
-     * Needs to set when push === true
+     * How should urls to this registry be labeled?
+     * ie DockerHub, ECR, etc (friendly name instead of big tag string)
+     * if not supplied, we'll display the tag
      */
+    label?: string;
+    username?: string;
```

Comment:
comments?

Reply:
This is part of the dockerRegistry interface: https://github.com/atomist/sdm-pack-docker/blob/ipcrm_multi_registry/lib/docker/executeDockerBuild.ts#L44-L79